python/sidecar/typing.py

```diff
@@ -202,7 +207,7 @@ class SearchResponse(RefStudioModel):
 
 
 class OpenAISettings(RefStudioModel):
-    api_key: str = ""
+    api_key: str = os.environ.get("OPENAI_API_KEY")
```

Comment:
This is where the key leaks -- when this schema gets converted to JSONSchema, the default value is serialized, leaking the key. TS doesn't have any notion of default values, so it goes away in `api-types.ts`.